content/guides/digging-deeper/route-model-binding.md

```diff
@@ -0,0 +1,279 @@
+
+```ts
+Route.get('/posts/:post/comments/:>comment', 'PostsController.show')
+```
+
+```ts
+import Post from 'App/Models/Post'
+import Comment from 'App/Models/Comment'
+import { bind } from '@adonisjs/route-model-binding'
+
+export default class PostsController {
+  @bind()
+  public async show({}, post: Post, comment: Comment) {
+    return { post, comment }
+  }
+}
+```
+
+For the above example to work, you will have to define the `comments` as a relationship on the `Post` model. The type of the relationship does not matter.
+
+```ts
+class Post extends BaseModel {
+  @hasMany(() => Comment)
+  public comments: HasMany<typeof Comment>
+}
+```
+
+The name of the relationship is looked up, converting the param name to `camelCase`. We will use both plural and singular forms to find the relationship.
+
+### Customizing relationship lookup
+By default, the relationship is fetched using the lookup key of the bound child model. Effectively the following query is executed.
+
+```ts
+await parent
+  .related('relationship')
+  .query()
+  .where(lookupKey, value)
+  .firstOrFail()
+```
+
+However, you can customize the lookup by defining the `findRelatedForRequest` method on the model (note, this is not a static method).
+
+```ts
+class Post extends BaseModel {
+  public findRelatedForRequest(ctx, param, value) {
+    /**
+     * Have to do this weird dance because of
+     * https://github.com/microsoft/TypeScript/issues/37778
+     */
+    const self = this as unknown as Post
+    const lookupKey = param.lookupKey === '$primaryKey' ? 'id' : param.lookupKey
+
+    if (param.name === 'comment') {
+      return self
+      .related('comments')
+      .query()
+      .where(lookupKey, value)
+      .firstOrFail()
+    }
+  }
+}
+```
+
+## Unbound params
+You will often have parameters that are raw values and cannot be tied to a model. In the following example, the `version` is a regular string value and not backed using the database.
+
+```ts
+Route.get(
+  '/api/:version/posts/:post',
+  'PostsController.show'
+)
+```
+
+You can represent the `version` as a string on the controller method, and we will perform no database lookup. For example:
+
+```ts
+import Post from 'App/Models/Post'
+import { bind } from '@adonisjs/route-model-binding'
+
+class PostsController {
+  @bind()
+  public async show({}, version: string, post: Post) {}
+}
+```
+
+Since the route params and the controller method arguments are matched in the same order they are defined, you will always have to type-hint all the parameters.
+
+## Compatibility with ESLint
+
+If ever you are using the rule `@typescript-eslint/consistent-type-imports`, you will notice that it will automatically replace your `import` with `import type`. This will completely break the route model binding, as the types will be removed at runtime, and so the `bind` decorator will be unable to retrieve the class from your model.
```

Comment:
```suggestion
If you use the rule `@typescript-eslint/consistent-type-imports`, you will notice that it will automatically replace your `import` with `import type`. Unfortunately, this will ultimately break the route model binding, as the types will be removed at runtime, so the `bind` decorator cannot retrieve the class from your model.
```